src/test/java/spoon/test/processing/ProcessingTest.java

```diff
@@ -41,6 +44,20 @@ public void testInsertBegin() throws Exception {
 							+ constructor.getSimpleName(), "int i = 0;",
 					constructor.getBody().getStatement(1).toString());
 		}
+			
+		CtConstructor<?> constructor = type.getConstructor(type.getFactory().Type().INTEGER_PRIMITIVE);
+    String myBeforeStatementAsString = "int before";
+    for (CtSwitch<?> ctSwitch : constructor.getElements(new TypeFilter<CtSwitch<?>>(CtSwitch.class))) {
+      ctSwitch.insertBefore(type.getFactory().Code()
+        .createCodeSnippetStatement(myBeforeStatementAsString));
+    }
+    assertEquals("insert has not been done at the right position", myBeforeStatementAsString, constructor.getBody().getStatement(3).toString());
+    assertEquals("insert has not been done at the right position", myBeforeStatementAsString, constructor.getBody().getStatement(5).toString());
```

Comment:
If a copy paste the constructor here:

```java
	public SampleForInsertBefore(int j) {
		this(j, 0);
		new Thread() {
		};
		switch (j) {
      default:
        break;
    }
    switch (j) {
      default:
        break;
    }
    switch (j) {
      default: {
        break;
      }
    }
```

We can see that the position 5 is a `break`.